src/main/java/Task.java

```diff
@@ -0,0 +1,98 @@
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+public class Task {
+
+    // all class variables private
+
+    protected String taskInfo;
+    protected boolean isDone;
+    protected String taskType;
+    protected Date by;
+    protected String byString;
+
+    public Task(String info, String type,String by) {
+        /**
+         *  constructor, saves info of task
+         *  @params String info: task information
+         * @params String type: type of task
+         * @return formatted string containing info and status of task
+         */
+        this.taskInfo = info;
+        this.isDone = false;
+        this.taskType = type;
+        if (!by.equals("")) {
+            this.by = convertToDate(by);
+        }
+        this.byString = by;
+    }
+    // getters
+    public String getTaskInfo() {
+        /**
+         *  getter of info of task
+         * @return String task info
+         */
+        return taskInfo;
+    }
+    public String getStatus() {
+        /**
+         *  getter of status of task
+         * @return tick or cross depending on isDone
+         */
+        // taken from partial soln tq prof
+        return (isDone ? "\u2713" : "\u2718");
+    }
```

Comment:
Clear comments, however, you might want to consider adding a linespace between your methods to improve readability?